src/fs/mem.rs

```diff
@@ -472,11 +474,40 @@ impl ObjectInterface for MemDirectoryInterface {
 
 #[derive(Debug)]
 pub(crate) struct MemDirectory {
-	inner:
-		Arc<RwLock<BTreeMap<String, Box<dyn VfsNode + core::marker::Send + core::marker::Sync>>>>,
+	inner: Arc<RwLock<MemDirectoryMap>>,
 	attr: FileAttr,
 }
 
+fn thin_tree_to_vfs_node(
+	thin_tree: ThinTree<'static>,
+	t: timespec,
+) -> io::Result<Box<dyn VfsNode + Send + Sync>> {
+	Ok(match thin_tree {
+		ThinTree::File { content, metadata } => {
+			let mut st_mode = AccessPermission::S_IRUSR;
+			if metadata.is_exec {
+				st_mode |= AccessPermission::S_IXUSR;
+			}
+			Box::new(RomFile::new_with_timestamp(content, st_mode, t))
+				as Box<dyn VfsNode + Send + Sync>
+		}
+		ThinTree::Directory(d) => Box::new(MemDirectory {
+			inner: Arc::new(RwLock::new(
+				d.into_iter()
+					.map(|(key, value)| Ok((key.to_string(), thin_tree_to_vfs_node(value, t)?)))
```

Comment:
This doesn't work because `ThinTree` filenames are `Vec<u8>`, not strings, which makes stuff a bit complicated.